src/magnify/registry.py

```diff
@@ -29,11 +32,254 @@ def component_factory(*args, **kwargs):
+    min_button_radius / max_button_radius : 
+        The minimum/maximum radius (in pixels) for detecting buttons in the image.
+    low_edge_quantile / high_edge_quantile : 
+        The lower/upper quantile for edge detection, used to identify the dimmest edges when detecting buttons.
+    num_iter : 
+        The maximum number of iterations to perform during the button detection process, controlling the depth of search.
+    min_roundness : 
+        The minimum roundness value for detected buttons. Buttons that do not meet this roundness threshold are excluded.
+    cluster_penalty : 
+        The penalty factor for clustering during button detection. This parameter helps control how strongly nearby buttons are grouped together.
+    roi_length : 
+        The length (in pixels) of the region of interest (ROI) around detected buttons.
+    progress_bar : 
+        If True, display a progress bar during processing to track the progress of the pipeline.
+    search_timestep : 
+        A list of time steps to focus on for button detection. If provided, only the specified time steps are used. If `None`, all time steps are used.
+    search_channel : 
+        The channel or list of channels to use for button detection and expression filtering. If `None`, all channels will be used. Channels are used to compute foreground and background values for contrast-based filtering.
+    squeeze : 
+        If True, removes singleton dimensions from the dataset(dimensions of size 1).
+    roi_only : 
+        If True, only returns the region of interest(roi) from the dataset, ignoring other parts of the image.
+    drop_tiles : 
+        If True, removes the "tile" variable from the dataset after stitching, effectively dropping individual tile data.
+
+    Returns
+    -------
+    Processed image(s): xr.Dataset | list[xr.Dataset]
+        The processed image in xr.Dataset from executed pipeline as the outcome of the image pipeline workflow.
+
+    Notes
+    -----
+    This function uses a pipeline architecture to process image data. The steps in the pipeline include:
+    
+    - 'identify_buttons' : Identifies buttons based on the provided `pinlist` or `shape` parameters and assigns valid markers.
+    - 'stitch' : Stitches image tiles based on the overlap parameter.
+    - 'rotate' : Rotates the stitched image by the specified angle.
+    - 'find_buttons' : Detects buttons based on edge detection and clustering.
+    - 'filter_expression' : Filters buttons based on foreground-background contrast differences using a minimum contrast threshold.
+    - 'filter_nonround' : Filters out buttons that do not meet the roundness criteria.
+    - 'filter_leaky' : Removes buttons that are determined to be leaky or poorly segmented.
+    - 'drop' : Optionally removes unnecessary tiles and simplifies the dataset based on the `squeeze`, `roi_only`, and `drop_tiles` options.
```

Comment:
Order these based on the order they execute in.

Reply:
Resolved. Now they are ranked based on the order that they are added to the pipeline